Crash when cisco anyconnect VPN connects or disconnects

### Summary

<!-- Please insert a one line summary of the issue below -->
In the past two weeks or so, I noticed Zed crashes when my VPN (cisco anyconnect) connects or disconnects.
<!-- Include all steps necessary to reproduce from a clean Zed installation. Be verbose -->
Steps to trigger the problem:
1. Open and use Zed as usual
2. Connect to VPN using the cisco anyconnect client
3. Zed crashes immediately after VPN connects (the Mac OS prompts something like "do you want to reopen Zed?")

Actual Behavior:
This kind of crash happens any time when my VPN status changes, either connect or disconnect. I assume some kind of network changes triggered this crash.
And also to be clear, when it crashes, I am not connecting to any remote files requiring VPN. Only local files were opened in Zed upon the crash.
This only started to happen in recent two weeks or so and I typically update my Zed to the latest version when it's available, so it's likely something introduced in recent versions of Zed or VPN client.

Expected Behavior:
Zed should behave normally regardless of my VPN status.

### Zed Version and System Specs

Zed: v0.177.8 (Zed) 
OS: macOS 15.3.2
Memory: 128 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

Sure, the relevant section from my `Zed.log` is attached below.

I also tried multiple versions:

* v0.173.11 -- same crash happened
* v0.171.5 -- no crash, work as expected

<details>
  <summary>Zed.log</summary>

```
2025-03-13T20:50:17.348892-04:00 [INFO] ========== starting zed ==========
2025-03-13T20:50:17.378337-04:00 [INFO] Opening main db
2025-03-13T20:50:17.379091-04:00 [INFO] Opening main db
2025-03-13T20:50:17.402293-04:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2025-03-13T20:50:17.463983-04:00 [INFO] No prompt template overrides directory found at ~/.config/zed/prompt_overrides. Using built-in prompts.
2025-03-13T20:50:17.464555-04:00 [INFO] extensions updated. loading 4, reloading 0, unloading 0
2025-03-13T20:50:17.49119-04:00 [INFO] Opening main db
2025-03-13T20:50:17.538884-04:00 [INFO] set status on client 0: Authenticating
2025-03-13T20:50:17.553172-04:00 [INFO] Opening main db
2025-03-13T20:50:17.555686-04:00 [INFO] set status on client 106536: Connecting
2025-03-13T20:50:17.558107-04:00 [INFO] Opening main db
2025-03-13T20:50:17.570207-04:00 [INFO] set environment variables from shell:/bin/zsh, path:~/opt/devpy3/bin:~/opt/bin:/snap/bin:~/Library/pnpm:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/opt/podman/bin:~/.cargo/bin
2025-03-13T20:50:18.060167-04:00 [INFO] Node runtime install_if_needed
2025-03-13T20:50:18.169689-04:00 [INFO] starting language server process. binary path: "~/Library/Application Support/Zed/node/node-v22.5.1-darwin-arm64/bin/node", working directory: "/", args: ["~/Library/Application Support/Zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2025-03-13T20:50:18.248521-04:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2025-03-13T20:50:18.546471-04:00 [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.236.0 initialized"
}
2025-03-13T20:50:18.546547-04:00 [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
{
  "items": [
    {
      "section": "github.copilot"
    },
    {
      "section": "github-enterprise"
    },
    {
      "section": "http"
    },
    {
      "section": "telemetry"
    }
  ]
}
2025-03-13T20:50:19.005858-04:00 [INFO] add_connection;
2025-03-13T20:50:19.006326-04:00 [INFO] set status on client 106536: Connected { peer_id: PeerId { owner_id: 699, id: 2256959 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2025-03-13T20:56:04.766917-04:00 [ERROR] connection error: delay between messages too long
2025-03-13T20:56:04.767514-04:00 [INFO] set status on client 106536: ConnectionLost
2025-03-13T20:56:04.768503-04:00 [INFO] set status on client 106536: Reauthenticating
2025-03-13T20:56:04.768969-04:00 [INFO] set status on client 106536: Reconnecting
2025-03-13T20:56:05.179733-04:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2025-03-13T20:56:05.749665-04:00 [INFO] add_connection;
2025-03-13T20:56:05.750345-04:00 [INFO] set status on client 106536: Connected { peer_id: PeerId { owner_id: 699, id: 2261903 }, connection_id: ConnectionId { owner_id: 0, id: 1 } }
2025-03-13T20:56:20.434291-04:00 [ERROR] {
  "thread": "<unnamed>",
  "payload": "illegal write_vectored return value",
  "location_data": {
    "file": "/Users/administrator/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/rustls-0.23.23/src/vecbuf.rs",
    "line": 197
  },
  "backtrace": [
    "backtrace::capture::Backtrace::create::h21b339776477734e+11438496",
    "backtrace::capture::Backtrace::new::hc8a089174c4cda25+11438304",
    "zed::reliability::init_panic_hook::{{closure}}::h4b50e587bad86e79+87129624",
    "std::panicking::rust_panic_with_hook::hc9e2ec2a12b10e12+68251940",
    "std::panicking::begin_panic_handler::{{closure}}::h3b6449e2eb83bf4c+68250728",
    "std::sys::backtrace::__rust_end_short_backtrace::h38d85099ddd107a3+68245136",
    "_rust_begin_unwind+68249940",
    "core::panicking::panic_fmt::hd4003258f6ab7b29+96358324",
    "rustls::vecbuf::ChunkVecBuffer::write_to::h8778c9a6db0dbb9e+65123500",
    "async_tls::rusttls::stream::Stream<IO>::complete_inner_io::hf7a6c77970e9735a+14191976",
    "<async_tungstenite::compat::AllowStd<S> as std::io::Write>::write::hb6c4cf7cee411cc7+14163688",
    "<futures_util::stream::stream::split::SplitSink<S,Item> as futures_sink::Sink<Item>>::poll_flush::hee658d33693ee90a+13065972",
    "rpc::proto::MessageStream<S>::write::{{closure}}::h2f09924d62dcfb08+13162644",
    "rpc::peer::Peer::add_connection::{{closure}}::h30096b88d355547d+13125948",
    "async_task::raw::RawTask<F,T,S,M>::run::hed79da3ea2c66943+13047884",
    "gpui::platform::mac::dispatcher::trampoline::h24d323e8ad57fa99+35301348",
    "_pthread_mach_thread_np+2411889564"
  ],
  "app_version": "0.177.8",
  "app_commit_sha": "e37aa899841ca8f4a3bd6c47c94045601b5bf734",
  "release_channel": "stable",
  "target": "aarch64-apple-darwin",
  "os_name": "macOS",
  "os_version": "15.3.2",
  "architecture": "aarch64",
  "panicked_on": 1741913780434,
  "system_id": "87c59e8c-fae3-47be-8803-8a25cf11464e",
  "installation_id": "e22ab944-fea2-4678-80c2-604a5d65c3c0",
  "session_id": "42d005e1-cb95-41ae-8966-778d3a279922"
}

```
</details>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -510,7 +510,7 @@ runtimelib = {  git = "https://github.com/ConradIrwin/runtimed", rev = "7130c804
 rustc-demangle = "0.1.23"
 rust-embed = { version = "8.4", features = ["include-exclude"] }
 rustc-hash = "2.1.0"
-rustls = { version = "0.23.22" }
+rustls = { version = "0.23.26" }
 rustls-platform-verifier = "0.5.0"
 scap = { git = "https://github.com/zed-industries/scap", rev = "08f0a01417505cc0990b9931a37e5120db92e0d0", default-features = false }
 schemars = { version = "0.8", features = ["impl_json_schema", "indexmap2"] }

